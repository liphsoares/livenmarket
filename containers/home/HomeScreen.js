import {
    Container, Header, Body, Text,
    Card, CardItem, View, Left, Right, Title, Toast, Thumbnail, Icon, Spinner
} from "native-base";
import React, { useEffect, useState } from 'react';
import { StyleSheet, Alert, FlatList, TouchableOpacity } from 'react-native';
import { useIsFocused } from '@react-navigation/native';
import CartIcon from "../sale/components/CartIcon";

export default function HomeScreen({ props, navigation }) {
    const isFocused = useIsFocused();
    const [searchResults, setSearchResults] = useState([]);
    const [IsLoading, setIsLoading] = useState(true);
    const [badge, setBadge] = useState(0);

    useEffect(() => {
        getData();
    }, [])

    useEffect(() => {
        if (isFocused) setBadge(CART_ITEMS.length);
    }, [props, isFocused]);

    async function getData() {
        setIsLoading(true);
        fetch('https://5d6da1df777f670014036125.mockapi.io/api/v1/product', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        }).then((data) => {
            data.json().then((res) => {
                if (res) {
                    let arr = res;
                    console.log(arr.length)
                    arr.sort((a, b) => { return b.name > a.name })
                    setSearchResults(arr);
                    setIsLoading(false);
                    setBadge(CART_ITEMS.length);
                }
            }).catch((err) => {
                Toast.show({
                    text: err
                }, 3000);
                setIsLoading(false);
            })
        }).catch((err) => {
            setIsLoading(false);

            Alert.alert(
                "Market",
                "Não consegui carregar o Catálogo de Produtos, deseja tentar novamente?",
                [
                    {
                        text: "SIM",
                        onPress: () => getData(),
                        style: "default",
                    },
                    {
                        text: "NÃO",
                        onPress: () => { },
                        style: "default",
                    },
                ]
            );
        });
    }

    return (
        <Container>
            <Header transparent androidStatusBarColor="#08012a" style={{ backgroundColor: '#08012a' }}>
                {/* <Left /> */}
                <Body>
                    <Title style={{ color: '#fff', textAlign: 'center' }}>Web Market</Title>
                </Body>
                <Right>
                    <TouchableOpacity
                        testID="button-cart"
                        onPress={() => {
                            navigation.navigate('Cart');
                        }}>

                        <CartIcon value={badge} />
                    </TouchableOpacity >
                </Right>
            </Header>

            {IsLoading ?
                <View
                    testID="loading"
                    style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                    <Spinner size={64} color="#ccc" />
                    <Text note>Aguarde, Carregando...</Text>
                </View>
                :
                <FlatList

                    data={searchResults}
                    renderItem={({ item }) => {
                        return (
                            <Card transparent testID="flat-list">
                                <CardItem bordered button={true} onPress={() => {
                                    navigation.navigate('DetailsItem', { product: item })
                                }}>
                                    <Left style={{ flex: 1 }}>
                                        {
                                            item.image !== '' ?
                                                <Thumbnail large source={{ uri: item.image.replace('http://lorempixel.com', 'https://loremflickr.com/') }} />
                                                : <Icon name='md-person' />
                                        }
                                        <Body>
                                            <Text>{item.name}</Text>
                                            <Text note>R$ {item.price}</Text>
                                            <Text style={{ marginTop: 10, fontSize: 10 }} note>Saldo: {item.stock}</Text>
                                        </Body>

                                    </Left>

                                    <Right style={{ flex: 0.1 }}>
                                        <Icon name="arrow-forward" size={16} />
                                    </Right>
                                </CardItem>
                            </Card>
                        )
                    }}
                    keyExtractor={item => String(item.id)}
                // ListFooterComponent={renderFooter}
                // onEndReached={this.handleLoadMore}
                // onEndReachedThreshold={5}
                />}

        </Container>
    )

}

const s = StyleSheet.create({
    CardStyle: {
        height: 200,
        backgroundColor: '#08012a',

        borderRadius: 16,
        marginHorizontal: 5
    },
    CardTextStyle: {
        marginTop: 20,
        color: '#78d98a',
        fontSize: 20
    },
    CardIconStyle: {
        color: '#fff'
    },
    image: {
        flex: 1,
        resizeMode: 'cover',
        justifyContent: 'center',
    }
})