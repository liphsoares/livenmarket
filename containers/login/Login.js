import React, { useState } from 'react';
import { Button, Container, View, Item, Input, Label, Header, Spinner } from 'native-base';
import { TouchableOpacity, Text, KeyboardAvoidingView, Platform,  StyleSheet, Image } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

export default function Login({ props, navigation }) {
    const [Username, setUsername] = useState('');
    const [Pwd, setPwd] = useState('');
    const [isSecureTextEntry, setisSecureTextEntry] = useState(true);
    const [IsLoading, setIsLoading] = useState(false);
    return (
        <Container style={{backgroundColor: "#08012a",}}>
            <Header transparent androidStatusBarColor="transparent" style={{ backgroundColor: 'transparent', height: 0 }} />
            <KeyboardAvoidingView style={{ flex: 1 }} behavior={Platform.OS === "ios" ? "padding" : null} enabled>

                <View style={{
                    flex: 1,
                    
                    alignItems: 'center',
                    flexDirection: 'column',
                    justifyContent: 'center',
                    backgroundColor: 'transparent'
                }} >
                    <Image style={{ width: 240, height: 80, bottom: 100 }} source={require('../../assets/images/logo-2x.png')} />
                    <View style={{
                        alignSelf: 'center',
                        alignContent: 'center',
                        width: '90%',
                        justifyContent: 'center',
                        flexDirection: 'column',
                        alignItems: 'center',
                        backgroundColor: '#fff',
                        padding: 30,
                        borderRadius: 16,
                        marginBottom: 50
                    }}>

                        <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
                            <Item floatingLabel style={{ width: '100%' }}>
                                <Label style={{ color: '#08012a' }}>Informe seu cpf ou cnpj</Label>
                                <Input
                                    style={{ color: '#08012a' }}
                                    keyboardType="number-pad"
                                    value={Username}
                                    onChangeText={(value) => {

                                    }}
                                />
                            </Item>
                        </View>
                        <View style={{ flexDirection: 'row', justifyContent: 'space-between', marginTop: 20 }}>
                            <Item floatingLabel style={{ width: '100%' }} >
                                <Label style={{ color: '#08012a' }}>Senha</Label>
                                <Input secureTextEntry={isSecureTextEntry}
                                    style={{ color: '#08012a' }}
                                    value={Pwd}
                                    onChangeText={(value) => { }}
                                    onSubmitEditing={() => {

                                    }}
                                />
                            </Item>
                            <Button style={{ right: 0, position: 'absolute' }} transparent onPress={() => {

                            }}>
                                <MaterialIcons
                                    name={isSecureTextEntry ? "visibility" : "visibility-off"} size={24} color="#fff" />
                            </Button>
                        </View>

                        <View style={{
                            flexDirection: 'row',
                            justifyContent: 'space-between',
                            marginTop: 40
                        }}>
                            <TouchableOpacity style={{ flex: 1 }} onPress={() => {
                                navigation.navigate('HomeScreen', { username: Username });
                            }}>
                                <Text style={{ color: '#08012a', textAlign: 'left', fontSize: 12 }}>Primeiro acesso</Text>
                            </TouchableOpacity>
                            <TouchableOpacity style={{ flex: 1 }} onPress={() => {
                                navigation.navigate('ForgotPassword', { username: Username });
                            }}>
                                <Text style={{ color: '#08012a', textAlign: 'right', fontSize: 12 }}>Esqueci minha senha</Text>
                            </TouchableOpacity>
                        </View>

                        {IsLoading ?
                            <Spinner color="#fff" />
                            : <Button full style={{ marginTop: 20, backgroundColor: "#78d98a" }}
                                onPress={() => {
                                   navigation.navigate('HomeScreen')

                                }}>
                                <Text style={{ color: '#08012a', fontSize: 20, fontWeight:'bold' }}>ENTRAR</Text>
                            </Button>}
                    </View>
                </View>

            </KeyboardAvoidingView>
        </Container >
    )
}

const styles = StyleSheet.create({

    image: {
        flex: 1,
        resizeMode: 'cover',
        justifyContent: 'center',
    }
});