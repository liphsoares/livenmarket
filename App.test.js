import React from 'react';
import Login from './containers/login/Login'
import {cleanup, render, fireEvent} from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import HomeScreen from './containers/home/HomeScreen';


afterEach(cleanup);

describe('Login', () => {

  it('O Texto do Botão  ENTRAR', () => {
    const defaultText = 'ENTRAR';
    const notFoundText = 'Not found text';

    const {toJSON, getByText, queryByText} = render(<Login />);

    const foundDefaultTextElement = getByText(defaultText);
    const notFoundTextElement = queryByText(notFoundText);

    expect(foundDefaultTextElement.props.children).toEqual(defaultText);
    expect(notFoundTextElement).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });
 
  it('Renderizar Botão Login', () => {
    const testIdName = 'btn-how-to-login';

    const {getByTestId} = render(<Login />);

    const foundButton = getByTestId(testIdName);

    expect(foundButton).toBeTruthy();
  });

});

describe('HomeScreen', () => {

  it('Renderizar o HomeSreen', () => {
    const home = render(<HomeScreen />);
    expect(home.toJSON()).toMatchSnapshot();
  });
 
});

