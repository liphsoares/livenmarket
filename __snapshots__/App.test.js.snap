// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login O Texto do Bot√£o  ENTRAR 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#08012a",
      },
    ]
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        androidStatusBarColor="transparent"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "paddingTop": undefined,
              "right": 0,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
              "top": 0,
            },
            Object {
              "backgroundColor": "transparent",
              "height": 0,
            },
          ]
        }
        transparent={true}
      />
    </RCTSafeAreaView>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "bottom": 100,
            "height": 80,
            "width": 240,
          }
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderRadius": 16,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 50,
              "padding": 30,
              "width": "90%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 2,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "left": 0,
                  "opacity": 1,
                  "paddingBottom": undefined,
                  "paddingTop": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 18,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#575757",
                      "fontSize": 17,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                    Object {
                      "color": "#08012a",
                    },
                  ]
                }
              >
                <Text
                  float={true}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#08012a",
                        "fontSize": 15,
                        "lineHeight": 30,
                      },
                    ]
                  }
                >
                  Informe seu cpf ou cnpj
                </Text>
              </Text>
            </View>
            <TextInput
              allowFontScaling={true}
              editable={true}
              keyboardType="number-pad"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder={null}
              placeholderTextColor="#575757"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingBottom": 7,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 3,
                    "top": 8,
                  },
                  Object {
                    "color": "#08012a",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value="dev"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 2,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "left": 0,
                  "opacity": 1,
                  "paddingBottom": undefined,
                  "paddingTop": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 18,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#575757",
                      "fontSize": 17,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                    Object {
                      "color": "#08012a",
                    },
                  ]
                }
              >
                <Text
                  float={true}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#08012a",
                        "fontSize": 15,
                        "lineHeight": 30,
                      },
                    ]
                  }
                >
                  Senha
                </Text>
              </Text>
            </View>
            <TextInput
              allowFontScaling={true}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder={null}
              placeholderTextColor="#575757"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingBottom": 7,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 3,
                    "top": 8,
                  },
                  Object {
                    "color": "#08012a",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value="dev"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "position": "absolute",
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 40,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#08012a",
                  "fontSize": 12,
                  "textAlign": "left",
                }
              }
            >
              Primeiro acesso
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#08012a",
                  "fontSize": 12,
                  "textAlign": "right",
                }
              }
            >
              Esqueci minha senha
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#78d98a",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 0,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
          testID="btn-how-to-login"
        >
          <Text
            style={
              Object {
                "color": "#08012a",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            ENTRAR
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
